src/test/java/com/vinyl/DeleteProductFromCartTest.java

```diff
@@ -0,0 +1,133 @@
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product, cart)).isNull();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartDoesNotExist() {
+        deleteCart();
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndTriesToDeleteFromAnotherCart() {
+        User otherUser = createUser("otheruser@email.com");
+        Cart otherCart = defaultEntitiesHelper.createCart(otherUser);
+        defaultEntitiesHelper.createProductCart(otherCart,product);
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+
+        ResponseEntity<?> response = trt.exchange("/users/" + otherUser.getId() + "/" + product.getId(), HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product,otherCart)).isNotNull();
+    }
+
+    @Test
+    public void testWhenProductNotInCart() {
+        deleteCart();
+        defaultEntitiesHelper.createCart(user);
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+
+        Product otherProduct = createOtherProduct();
+        ResponseEntity<?> response = trt.exchange("/users/" + user.getId() + "/" + otherProduct.getId(), HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductIsInvalid() {
+        deleteCart();
+        defaultEntitiesHelper.createCart(user);
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+        Integer invalidProductId=000;
+
+        ResponseEntity<?> response = trt.exchange("/users/" + user.getId() + "/" + invalidProductId, HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+
+    private Product createOtherProduct() {
+        Product product = new Product();
+        product.setArtist("MJ");
+        product.setCategory("Pop Music");
+        product.setDescription("Best Seller");
+        product.setPrice(80.5);
+        product.setProductName("MJ Vinyl");
+        product.setStock(14);
+
+        return productRepository.save(product);
+    }
+
+    private Cart cartSetup() {
+        Cart cart = defaultEntitiesHelper.createCart(user);
+        ProductCart pc = defaultEntitiesHelper.createProductCart(cart, product);
+        cart.setProducts(Lists.newArrayList(pc));
+
+        return cartRepository.save(cart);
+    }
+
+    private void deleteCart() {
+        productCartRepository.deleteAll();
+        productRepository.deleteAll();
```

Comment:
don't delete all products just for deleting a cart :dagger:

Reply:
fixed